### DISCUSSION TITLE
Compiling ORSSerialPort on Catalina en XC12 errors

### DISCUSSION THREAD

**Discussion by lordAnubis** [armadsen/ORSSerialPort#168] [Q&A]
I did move it from the General area to Q&A. Hope you don't mind. I guess it should be here.

I am busy to write code for communicating MacOS using Swift5 with PIC's and Arduino's ( Nano Every ).

However the docs are very clear,but  I can't get it compiled. It give's me 20 errors.
So did step back to try to the samples, but again same error's
Then try to compile ORSSerialPort.framework and the same 20 ARC? error's.

I do get f.e

ORSSerialPort.m:150:37: No known class method for selector 'deviceFromBSDPath:'
Incompatible pointer to integer conversion initializing 'io_object_t' (aka 'unsigned int') with an expression of type 'id'
/Users/rva/OnlineDocumenten/Dropbox/Projecten/_Develop/_Projects-Apple_CodeDev/_Projects-Cocoa/SWIFT-

ORSSerialPort.m:278:8: No visible @interface for 'ORSSerialPort' declares the selector 'updateModemLines'
I am using XC12.3 and Catalina as build target os.

In the Framework project is there something that I have to add, or that I didn't find in the docs?
BTW I did add the -fobjc-arc compiler flag at OSRSerial>Build Phases> compile sources OSRSerialPort.m and OSSerialPortManager.m to check if this was the crulpit.

Any suggestions?

**Comment by flemingm** [comment]
New Xcode is doing a “better job” of type checking.  I have not tried this project with latest Xcode yet but suggest where to start:

Ie
> 
> : No known class method for selector 'deviceFromBSDPath:'
> Incompatible pointer to integer conversion initializing 'io_object_t' (aka 'unsigned int') with an expression of type 'id'
> 

And
> 
> No visible @interface for 'ORSSerialPort' declares the selector 'updateModemLines'
> 
> 
> 

OS it likely need come casting or interface updates

Missing symbols most like missing library  



Mark Fleming

**Reply by lordAnubis** [reply]
I did import as an extra the IOKit or do you think about another framework/library?

**Reply by flemingm** [reply]
most likely IOKit is only thing needed - not actively working on using this library right now so project are on backups not accessible right now.

**Comment by lordAnubis** [comment]
Hi Mark thank you for your reply.
If I understand it clearly then to keep older XC/Swift versions happy there should be 
#if compiler(>=5) && swift(<5) e.t.c tests be included

If I compile the OSRSerialPort framework with an older version of XC then will it work together with my current project if I compile my current project with Swift5/XC12?
